fiftyone/utils/utils3d.py

```diff
@@ -0,0 +1,394 @@
+"""
+3D utilities.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import itertools
+import logging
+from typing import Tuple
+from typing_extensions import Literal
+
+import eta.core.image as etai
+import numpy as np
+
+import fiftyone.core.collections as foc
+import fiftyone.core.fields as fof
+import fiftyone.core.media as fom
+import fiftyone.core.sample as fos
+import fiftyone.core.utils as fou
+import fiftyone.core.validation as fov
+from fiftyone.core.odm import DynamicEmbeddedDocument
+
+o3d = fou.lazy_import("open3d", callback=lambda: fou.ensure_package("open3d"))
+
+logger = logging.getLogger(__name__)
+
+Vec2Tuple = Tuple[int, int]
+Vec3Tuple = Tuple[int, int, int]
+
```

Comment:
Typically const vars are given all CAPS names, so SHADING_GRADIENT_MAP. Also, why the specific values 0.111, 0.222, etc.? What does 1/9 have to do with shading?

Reply:
Side comment: Ben started adding type hints to App server code, but none of `fiftyone.core` or `fiftyone.utils` have type hints currently, so I'd rather just omit them from here for consistency.

Reply:
@sashankaryal wrt the gradient map, we have [this function](https://github.com/voxel51/fiftyone/blob/dfb81b6fffb8ef9bb36bcb0d534f357e664ccfc8/fiftyone/core/config.py#L377) that can generate a colormap with any number of discrete colors and can support a variety of custom colorscales as documented there. I'd recommend leveraging that so user can customize the colorscale in a familiar way.

Reply:
Also, can you update so this method return images represented as `np.uint8` arrays in `[0, 255]`. In general, throughout the library, if we say we return an "image", we generally mean uint8, rather than float, especially when you're already discretizing.